app/scripts/migrations/092.3.ts

```diff
@@ -0,0 +1,52 @@
+import { isNullOrUndefined } from '@metamask/utils';
+import { cloneDeep } from 'lodash';
+
+type VersionedData = {
+  meta: { version: number };
+  data: Record<string, unknown>;
+};
+
+export const version = 92.3;
+
+/**
+ * This migration does the following:
+ *
+ * - Deletes currently stored advancedGasFee in preferences controller, replacing the default with an empty object
+ *
+ * @param originalVersionedData - Versioned MetaMask extension state, exactly what we persist to dist.
+ * @param originalVersionedData.meta - State metadata.
+ * @param originalVersionedData.meta.version - The current state version.
+ * @param originalVersionedData.data - The persisted MetaMask state, keyed by controller.
+ * @returns Updated versioned MetaMask extension state.
+ */
+export async function migrate(
+  originalVersionedData: VersionedData,
+): Promise<VersionedData> {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  migrateData(versionedData.data);
+  return versionedData;
+}
+
+function migrateData(state: Record<string, unknown>): void {
+  changeShapeAndRemoveOldAdvancedGasFeePreference(state);
+}
+
+function changeShapeAndRemoveOldAdvancedGasFeePreference(
+  state: Record<string, any>,
+) {
+  if (isNullOrUndefined(state.PreferencesController)) {
+    return;
+  }
+
+  state.AppStateController = {
+    ...(state.AppStateController ?? {}),
```

Comment:
Nit: Perhaps we should validate that `state.AppStateController` and `state.PreferencesController` are objects first? If it's some non-nullish unexpected type, this might make a mess of this state.

Reply:
We could also add validation of `advancedGasFee` property exists in `state.PreferencesController`?

Reply:
@Gudahtt should I just skip setting the trait entirely if its some non-nullish unexpected type?

Reply:
I guess so, yeah. The pattern in other migrations for when we encounter invalid state is to call `captureException` and return state unchanged, halting the migration

Reply:
Unless i'm mistaken... in 094.ts we use captureException, but we still modify the meta.version number? That seems like it could cause further breakages if we make future migrations that rely upon the state that 094 migrated us to.

Reply:
captureException doesn't halt execution right?

Reply:
That's correct, it doesn't halt execution. And also you are correct that it could affect later migrations, but those later migrations should also perform adequate checks and bail in the same way if corruption is detected (where "corruption" might mean "a previous required migration was not run").

Reply:
Oh also, the intent behind capturing this error is that usually if such a scenario would occur, we'd discover it in Sentry and fix it before much time has passed.